### ISSUE TITLE
Report if empty PropertyPatternClause could be written as declaration pattern within 'if' condition

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#1719]
We should report in case an `if` condition contains an IsPattern with a recursive pattern that is an empty PropertyPattern.

Following should trigger a warning:

```c#
if (someItem.SomeValue is { } value)
```

It should be re-written (using a declaration pattern) as:
```c#
if (someItem.SomeValue is SomeType value)
```

The latter one is easier to read and understand, also for those developers that are not so familiar with C# code.
